src/test/java/org/ballerinalang/client/HttpTestClient.java

```diff
@@ -0,0 +1,249 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InterruptedIOException;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import javax.net.ssl.SSLException;
+
+/**
+ * HttpRequest sending client
+ */
+public class HttpTestClient {
+
+    private String host;
+    private CloseableHttpClient httpclient;
+    private RequestConfig requestConfig;
+    private boolean isChunked = false;
+    private boolean isContentEncoded = false;
+    private String contentType = "text/plain";
+    private String charSet = "UTF-8";
+    private HashMap<String, String> requestHeaders = new HashMap<>();
+
+    public HttpTestClient(String host) {
+
+        if (host.endsWith("/")) {
+            this.host = host.substring(0, host.length() - 1);
+        } else {
+            this.host = host;
+        }
+
+        // Timeout configs
+        requestConfig = RequestConfig.custom().setSocketTimeout(10000).setConnectTimeout(10000).build();
+
+        // Handler for managing reties
+        HttpRequestRetryHandler conRetryHandler = (exception, executionCount, context) -> {
+            if (executionCount >= 3) {
+                // Do not retry if over max retry count
+                return false;
+            }
+            if (exception instanceof InterruptedIOException) {
+                // Timeout
+                return false;
+            }
+            if (exception instanceof UnknownHostException) {
+                // Unknown host
+                return false;
+            }
+            if (exception instanceof ConnectTimeoutException) {
+                // Connection refused
+                return true;
+            }
+            if (exception instanceof SSLException) {
+                // SSL handshake exception
+                return false;
+            }
+            HttpClientContext clientContext = HttpClientContext.adapt(context);
+            org.apache.http.HttpRequest request = clientContext.getRequest();
+            boolean idempotent = !(request instanceof HttpEntityEnclosingRequest);
+            if (idempotent) {
+                // Retry if the request is considered idempotent
+                return true;
+            }
+            return false;
+        };
+
+        httpclient = HttpClients.custom().setRetryHandler(conRetryHandler).disableContentCompression().build();
+    }
+
+    public String getCharSet() {
+        return charSet;
+    }
+
+    public void setCharSet(String charSet) {
+        this.charSet = charSet;
+    }
+
+    public void setChunked(boolean chunked) {
+        isChunked = chunked;
+    }
+
+    public void setContentEncoded(boolean contentEncoded) {
+        isContentEncoded = contentEncoded;
+    }
+
+    public void setContentType(String contentType) {
+        this.contentType = contentType;
+    }
+
+    public void setHeader(String header, String value) {
+        this.requestHeaders.put(header, value);
+    }
+
+    /**
+     * @param resourcePath
```

Comment:
If putting the Javadoc comment, it must be defined properly. The main comment must be here, and parameter definitions and the return value. Fix in all the places.

Reply:
This is fixed with #10